dvl-a50/dvl.py

```diff
@@ -334,26 +360,39 @@ def handle_velocity(self, data):
     def handle_position_local(self, data):
         if self.should_send == MessageType.POSITION_ESTIMATE:
             x, y, z, roll, pitch, yaw = data["x"], data["y"], data["z"], data["roll"], data["pitch"], data["yaw"]
-            timestamp = data["ts"]
+            self.timestamp = data["ts"]
             self.mav.send_vision_position_estimate(
-                    timestamp,
+                    self.timestamp,
                     [x, y, z],
-                    [roll, pitch, yaw])
+                    [roll, pitch, yaw],
+                    reset_counter=self.reset_counter)
+
+    def check_temperature(self):
+        now = time.time()
+        if now - self.last_temperature_check_time < 30:
+            return
+        self.last_temperature_check_time = now
+        try:
+            status = json.loads(request(f"http://{self.hostname}/api/v1/about/status"))
+
+            temp = float(status["temperature"])
+            if temp > 45.0:
```

Comment:
It would be nice to have in the class variables for temperature_check_time and temperature_threshold

Reply:
yep!